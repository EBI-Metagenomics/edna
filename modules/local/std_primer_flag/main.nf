
process STD_PRIMER_FLAG {
    // Check for presence of standard library of primers (stored in ./data/standard_primers)
    tag "$meta.id"
    label 'very_light'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mi-pimento:1.0.0--pyhdfd78af_0":
        "biocontainers/mi-pimento:1.0.0--pyhdfd78af_0"}"
    input:
    tuple val(meta), path(reads_merged)
    path(std_primer_library)

    output:
    tuple val(meta), path("*std_primers.fasta"), emit: std_primer_out
    path "*std_primer_out.txt"
    path "versions.yml"                        , emit: versions

    script:
    def std_primer_library_arg = "${std_primer_library}" ? "-p ${std_primer_library}" : ""
    print(std_primer_library)

    """
    pimento std -i ${reads_merged} ${std_primer_library_arg} -o ${meta.id}_${meta.var_region}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mi-pimento: \$( pimento --version | cut -d" " -f3 )
    END_VERSIONS
    """

}
