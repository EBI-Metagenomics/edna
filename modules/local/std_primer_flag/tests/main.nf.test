nextflow_process {

    name "Test Process STD_PRIMER_FLAG"
    script "../main.nf"
    process "STD_PRIMER_FLAG"
    tag "modules"
    tag "std_primer_flag"

    test("Should take a fastq.gz file and search for any hits to a list of standard primers, returning a fasta file with identified primers") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false, var_region:'16S-V3-V4' ], // meta map
                    file('${projectDir}/modules/local/std_primer_flag/tests/data/test.fastq.gz', checkIfExists: true),
                ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.std_primer_out).match("std_primer_out") }
            )
            
        }

    }

}
