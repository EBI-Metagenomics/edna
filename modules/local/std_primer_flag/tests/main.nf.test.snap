{
    "std_primer_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false,
                        "var_region": "16S-V3-V4"
                    },
                    "test_16S-V3-V4_std_primers.fasta:md5,f26cdc22a5231d2dbd0831f1f1983dc6"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-25T11:27:51.903776596"
    }
}